### DISCUSSION TITLE
How to install `irobot_create_msgs`?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#251] [Questions] [answered]
I noticed that irobot_create_msgs repository has releases, and ROS2 programmers wishing to interface to a particular Create3 need a procedure/tutorial for setting up their ROS2 workspace with the irobot_create_msgs release corresponding to their targeted Create3 "version" 

The Create3 "About" page of the webserver of my Create3 shows:
```
>>>>>>> system version
G.3.1
```

I am guessing that rather than cloning the current Galactic irobot_create_msgs repository to my workspace, I need the matching irobot_create_msgs "release".  (These appear to be compressed tar files - I have not worked with GitHub releases before).

What is the process I should follow now to setup my ROS2 (TurtleBot4 and) Create3 development workspace, and again if I choose to update my Create3 firmware in the future?

(Discussion https://github.com/iRobotEducation/create3_docs/discussions/240 shows two users that have hit this mismatch recently)

**Comment by alsora** [comment, marked as answer]
Hi @slowrunner,
most developers shouldn't use the github repository for [irobot_create_msgs repository](https://github.com/iRobotEducation/irobot_create_msgs).

The `irobot_create_msgs` should be installed from debian packages using the following command.
```
sudo apt-get update
sudo apt-get install ros-galactic-irobot-create-msgs
```

---- 

If you really need to build the `irobot_create_msgs` (the only reasons I can think of is if you are cross-compiling the code or building the code on a non-officially supported platform), then you should check which version you need in the Create 3 Docs Firmware release pages.
For example, you can see that G.3.1 uses version 1.2.4 of the messages https://iroboteducation.github.io/create3_docs/releases/g_3_1/

then you can clone the right version and build it using
```
mkdir -p ~/ws/src
git clone -b 1.2.4  https://github.com/iRobotEducation/irobot_create_msgs.git ~/ws/src/irobot_create_msgs
cd ~/ws
colcon build
```

**Reply by slowrunner** [reply]
My particular combination of following the TurtleBot4 getting started manual, and the Create3 doc pages seems to have confused me and that workspace. I have rebuilt with the following steps and my TurtleBot4/Create3 client node builds and runs successfully:

1) ==== Install TurtleBot 4 packages
```
sudo apt update
sudo apt install ros-galactic-turtlebot4-description \
ros-galactic-turtlebot4-msgs \
ros-galactic-turtlebot4-navigation \
ros-galactic-turtlebot4-node
```
2) ==== Install TurtleBot 4 Desktop
```
sudo apt install ros-galactic-turtlebot4-desktop
```

3) ==== Install TurtleBot 4 Simulator

* Install Dev Tools (probably already present)
```
sudo apt install -y \
python3-colcon-common-extensions \
python3-rosdep \
python3-vcstool
```

* Install sim packages
```
sudo apt install ros-galactic-turtlebot4-simulator ros-galactic-irobot-create-nodes
```

4) === Setup ROSDEP
```
sudo rosdep init
rosdep update
```

5) === BULDING A TURTLEBOT4 Node (My node is WaLI: Wall follower Looking for Intelligence)

make sure to include in ```<workspace>/src/<package>/package.xml```:  
```
  <depend>rclpy</depend>
  <depend>irobot_create_msgs</depend>
```
- run rosdep:  
```
rosdep install --from-paths src -y --ignore-src  
```
- do build  
```
colcon build --symlink-install --packages-select wali  
```
- add to environment  
```
source ~/tb4desk/install/setup.bash  
```
- run the node:  
```
ros2 run wali wali_node
```

This is the log of WaLi node running successfully today:
```
2022-11-23 22:03|------------ boot ------------
2022-11-24 06:58|wali_node.py| WaLI node started
2022-11-24 06:59|wali_node.py| ** WaLI Undocking: success at battery 99% **
2022-11-24 08:42|wali_node.py| ** WaLi Docking: success at battery 24% **
2022-11-24 08:54|lifelogger.dEmain| execution: 10.85
```

**Comment by slowrunner** [comment]
Closing - solved and stale.
